dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultDataQueryService.java

```diff
@@ -309,6 +310,22 @@ public List<OrganisationUnit> getUserOrgUnits(DataQueryParams params, String use
     return units;
   }
 
+  /**
+   * Retrieve the list of organizational units to which the current user has access rights.
+   *
+   * @param currentUser {@link User}
+   */
+  private List<OrganisationUnit> getOrganisationUnitsGrantedForAnalyticsData(User currentUser) {
+    Set<OrganisationUnit> organisationUnits = currentUser.getDataViewOrganisationUnits();
+    if (organisationUnits != null && !organisationUnits.isEmpty()) {
+      return organisationUnits.stream().sorted().toList();
+    } else {
+      // If the user has no analytics permissions for any organizational unit,
+      // apply their data capture permissions instead.
+      return currentUser.getOrganisationUnits().stream().sorted().toList();
```

Comment:
Better check if `currentUser.getOrganisationUnits()` is null, just in case.

Reply:
done